library/ssl_tls.c

```diff
@@ -4917,8 +4916,7 @@ int mbedtls_ssl_parse_sig_alg_ext( mbedtls_ssl_context *ssl,
                                     sig_alg,
                                     mbedtls_ssl_sig_alg_to_str( sig_alg ) ) );
 
-        if( ! mbedtls_ssl_sig_alg_is_supported( ssl, sig_alg ) ||
-            ! mbedtls_ssl_sig_alg_is_offered( ssl, sig_alg ) )
+        if( ! mbedtls_ssl_sig_alg_is_supported( ssl, sig_alg ) )
```

Comment:
I think for TLS 1.3 the "is_supported" check should be removed from here as well. And I'd say no need to add it in `ssl_tls13_select_sig_alg_for_certificate_verify` as we have `mbedtls_ssl_tls13_sig_alg_for_cert_verify_is_supported` there. But for TLS 1.2 we should keep both tests here. `ssl->tls_version` contains the negotiated version at this point and could be used to differentiate between TLS 1.2 and TLS 1.3.